### ISSUE TITLE
Game booting process doesn't use all threads available when RPCS3 Scheduler is selected on Thread Scheduler settings (Linux specific)

### ISSUE THREAD

**Issue by kpqi5858** [RPCS3/rpcs3#15578]
### Quick summary

This is Linux specific issue.

When RPCS3 Scheduler or RPCS3 Alternative Scheduler is selected on Thread Scheduler settings, the game booting process (Compiling PPU modules, Building SPU cache, etc) doesn't use all threads available, making it considerably slower.

### Details

On pthread_create(3) man page, it specifically mentions that child threads inherit the affinity mask of the calling thread. This causes the undesired behavior.

> **Linux-specific details**
> The new thread inherits copies of the calling thread's capability sets (see [capabilities(7)](https://man7.org/linux/man-pages/man7/capabilities.7.html)) and CPU affinity mask (see [sched_setaffinity(2)](https://man7.org/linux/man-pages/man2/sched_setaffinity.2.html)).

### Attach a log file

N/A

### Other details

kd-11 said some words about this [on discord](https://discord.com/channels/272035812277878785/272083901277667342/1224513039542259732)

**Comment by digant73** [comment]
@kpqi5858 is that still an issue? Here, probably @elad335 can eventually provide a feedback

**Comment by kpqi5858** [comment]
Yes, this issue still persists in 0.0.38-18263-3c401e7b. My cpu is Ryzen 9 5900HX (8c,16t)

**Comment by nergdron** [comment]
just ran into this as well on Arch, with `rpcs3-bin 0.0.40.19125-1`. when I switch to OS scheduler, the first batch of SPU cache generation for a big game with lots of modules seems to run in parallel, but when it gets to a big one that takes a long time (in my case, GT5, SPU cache module #673/2741), it blocks on the big module and doesn't progress with building more in parallel, just locks a single thread at 100% while it works. so there may also be logic errors or other things going on which limit the amount of parallelism it'll do here.

in any case, switching off the RPCS3 schedulers does seem like an improvement.

**Comment by nergdron** [comment]
ahhh! switching off mega spu blocks fixed my hang in that shader, and things progress in parallel the way you'd expect. so don't mind me, I just made a change that broke that specific game which led to it not building properly with OS scheduling.

**Comment by digant73** [comment]
@kpqi5858 please an update on current build

**Comment by kpqi5858** [comment]
Still an issue in RPCS3 v0.0.40-19392-0a5b88ac
